### ISSUE TITLE
Support pulsar2.11.0

### ISSUE THREAD

**Issue by xuanyangyang** [apache/pulsar-manager#509]
pulsar2.11.0 use jdk17.When is jdk17 supported?

**Comment by bpereto** [comment]
there is no dependency on jdk17?
But you are right, jdk-8 is ancient.

I builded the docker image with `FROM openjdk:17-slim-bullseye` and needed an extra java opts: `JAVA_OPTS: "--add-opens java.base/java.lang=ALL-UNNAMED"`

with this setup it works fine.

**Comment by tisonkun** [comment]
Yeah. This issue looks like to ask to bump the JDK version to 17 in this project.

You're welcome to submit a patch for doing so, or wait for other volunteer take it over :D

**Comment by onobc** [comment]
This issue came up today at the Pulsar community meeting, in the context of the Pulsar Helm issue w/ a related issue (using older Pulsar client and dependencies). 

I would like to bring everything up to the latest dependencies:
- Spring 6.1.3
- SpringBoot 3.2.2
- Pulsar 3.1.2

which implies Java 17. 

These changes will warrant a "major" version bump. I see the current version is still < `1.0.0`. 
We can figure out where the changes land, but likely it would be in a `1.0.0`.
